### PR TITLE
Add of support for 'unique' method on array wrapper

### PR DESCRIPTION
**Pull request by akinsella**
Hi,

I added some support for 'unique' method on array wrapper.

Here is a sample:

``` objective-c
        NSURL *fbReposUrl = [NSURL URLWithString:@"https://api.github.com/orgs/facebook/repos"];
        NSData *data = [NSData dataWithContentsOfURL:fbReposUrl];
        NSArray *repos = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:NULL];

        NSArray *languages = _array(repos).pluck(@"language").reject([Underscore isNull]).unique.unwrap;

        NSLog(@"Unique languages used by FaceBook: %@", languages);
```

### CODE REVIEW

File: Underscore/USArrayWrapper.m [hunk 1]
```diff
@@ -206,6 +206,21 @@ - (USArrayWrapper *)shuffle;
     };
 }
 
+
+- (USArrayWrapper *)unique;
+{
+    __block NSArray *(^unique)(NSArray *) = ^NSArray *(NSArray *input) {
```
Comment (robb) [inline Underscore/USArrayWrapper.m:212]: This block in not necessary.
I guess you based this on the `flatten` implementation but there it's just for recursion :-)

Comment (akinsella) [inline Underscore/USArrayWrapper.m:212]: Hi, 

My mistake. Yes, I based the change on flatten implementation. Do I need to resubmit  the pull request ?

**Comment by robb** [comment]
Hi Alexis.

Would you mind addressing my comment and create a new pull request against the development branch?

That branch contains a couple of syntax changes, most notably it adds a functional style for calling methods:

``` objective-c
NSArray *myArray = @[@1, @2, @2, @3, @1];
NSArray *uniques = Underscore.unique(myArray);
```

I'm a bit behind with pushing version 0.2, sorry for that!

Thanks!

File: Underscore/USArrayWrapper.m (hunk 1 above)
Comment (robb) [inline Underscore/USArrayWrapper.m:212]: While you can just push to this branch to update the PR, I don't think it makes sense to target development from here.

I suggest you reimplement it based on development and submit a new PR. Sorry for the trouble.

**Comment by akinsella** [comment]
Ok,
I'll do it right now.

**Comment by robb** [comment]
Cool, I close this Pull Request then

**Comment by robb** [comment]
Also, let's call it `uniq` to be in sync with Haskell, Underscore.js and Unix :-)

**Pull request closed without merging**